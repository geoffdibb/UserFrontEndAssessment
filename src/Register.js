import React, { Component } from 'react';
import axios from 'axios';


export default class Register extends Component {

    constructor() {
        super();
        this.state = {
            data: ""
        }
    }

    makeRequest = (event) => {
        event.preventDefault();
        let newItem = {
            username: event.target[0].value,
            email: event.target[1].value,
            password: event.target[2].value,
            password2: event.target[3].value
        }

        axios
            .post("http://localhost:5000/User/createUser", newItem)
            .then(response => {
                this.setState({
                    data: response.data
                });
                // window.alert(this.state.data.password, this.state.data.username, this.state.data.email)
            }).catch(err => console.log(err));
            
    };


    checkpass = () => {
  if (this.state.data.password==="Passwords do not match") {

    return <div>{this.state.data.password}</div>
  }
}
    render() {
        return (

            <div >
                    
                <h2>Register a new account </h2>
                <form onSubmit={this.makeRequest}>

                    <br></br>
                    <label for="Username">Username: </label>
                    <input type="text" id="Username" name="Username" required />
                    <br></br>

                    <label for="Email">Email: </label>
                    <input type="Email" id="Email" name="Email" required />
                    <br></br>

                    <label for="Password">Password: </label>
                    <input type="Password" id="Password" name="Password" required />
                    <br></br>

                    <label for="Password2">Confirm Password: </label>
                    <input type="Password" id="Password2" name="Password2" />
                    <br></br>

                    <input type="submit" value="RegisterAccount" />
                    <p id="systemResponse"> </p>
                </form>

            {this.checkpass()}
                <br />
                {this.state.data.email}

            </div>
        );
    }
}
