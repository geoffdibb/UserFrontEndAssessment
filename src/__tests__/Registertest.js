import React from 'react';
import ReactDOM from 'react-dom';
import Register from '../Register';
import renderer from 'react-test-renderer';


it('snapshot Register', () => {
  const tree = renderer.create(<Register/>).toJSON()
  expect(tree).toMatchSnapshot();
})