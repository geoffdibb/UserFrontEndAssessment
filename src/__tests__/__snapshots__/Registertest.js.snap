// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Register 1`] = `
<div>
  <h2>
    Register a new account 
  </h2>
  <form
    onSubmit={[Function]}
  >
    <br />
    <label
      for="Username"
    >
      Username: 
    </label>
    <input
      id="Username"
      name="Username"
      required={true}
      type="text"
    />
    <br />
    <label
      for="Email"
    >
      Email: 
    </label>
    <input
      id="Email"
      name="Email"
      required={true}
      type="Email"
    />
    <br />
    <label
      for="Password"
    >
      Password: 
    </label>
    <input
      id="Password"
      name="Password"
      required={true}
      type="Password"
    />
    <br />
    <label
      for="Password2"
    >
      Confirm Password: 
    </label>
    <input
      id="Password2"
      name="Password2"
      type="Password"
    />
    <br />
    <input
      type="submit"
      value="RegisterAccount"
    />
    <p
      id="systemResponse"
    >
       
    </p>
  </form>
  <br />
</div>
`;
